### PR TITLE
python311Packages.highdicom: init at 0.22.0

### PR DESCRIPTION
**Pull request by bcdarwin**
python311Packages.pillow-jpls: init at 1.3.2

## Description of changes

Init `highdicom`, a package for handling DICOM abstractions in Python.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/pillow-jpls/default.nix [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  rev = "c4c0155e47b17d5a2ab09d81256041eac8fd35b9";
+  src = fetchFromGitHub {
+    owner = "planetmarshall";
+    repo = "pillow-jpls";
+    inherit rev;
+    hash = "sha256-+cLd04t+gCAy0gxXi5lex368Y6/Vu/fLsu4E8zcG654=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml --replace '"conan~=2.0.16",' ""
```
Comment (zeuner) [inline pkgs/development/python-modules/pillow-jpls/default.nix:39]: Could you use `pythonRelaxDeps` or `pythonRemoveDeps` for this (see https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonrelaxdepshook-using-pythonrelaxdepshook)?

**Review by zeuner — COMMENTED** [review]
Didn't run it myself yet, but apart from the aforementioned, it looks reasonably done.

File: pkgs/development/python-modules/pillow-jpls/default.nix (hunk 1 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/pillow-jpls/default.nix:38]: Unfortunately pythonRemoveDeps doesn't appear to work (this has largely been my experience with pyproject-based packages, but I haven't really investigated why).

**Comment by bcdarwin** [comment]
Bumped `pillow-jpls` from `unstable-...` to a new release version.

File: pkgs/development/python-modules/pillow-jpls/default.nix (hunk 1 above)
Comment (zeuner) [reply pkgs/development/python-modules/pillow-jpls/default.nix:38]: Yes, I also noticed the `pythonRelaxDepsHook` does not always work. Just thinking it should be applied where already possible.

**Review by zeuner — APPROVED** [review]

**Review by marsam — APPROVED** [review]

**Comment by superherointj** [comment]
Since this PR landed, I'm getting this error:

> error: evaluation aborted with the following error message: 'lib.customisation.callPackageWith: Function called without required argument "setuptools_scm" at /home/intj/.cache/nixpkgs-review/pr-287792-3/nixpkgs/pkgs/development/python-modules/pillow-jpls/default.nix:20, did you mean "setuptools-scm"?'

**Pull request merged**